### PR TITLE
Implementation of ArmorStands

### PR DESCRIPTION
**Pull request by kaenganxt**
This pull requests implements armor stands and covers the following aspects:
- Correct placement of ArmorStands
- Sending of ArmorStand data to the client
- Destroying and interacting with the ArmorStand
- Saving and loading vanilla nbt data for the ArmorStand
- Implementation of Bukkit ArmorStand methods
- Implementation of the following events: PlayerInteractAtEntityEvent, PlayerArmorStandManipulateEvent

Currently the DisabledSlots nbt tag is not supported since bukkit does not provide api methods to manipulate it properly. So I would recommend adding it in a later PR after thinking about an implementation.

### CODE REVIEW

File: src/main/java/net/glowstone/entity/objects/GlowArmorStand.java [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+        for (int i = 0; i < 2; i++) {
+            pose[i] = EulerAngle.ZERO;
+        }
+        double ten = Math.toRadians(10);
+        double fifteen = Math.toRadians(15);
+        double one = Math.toRadians(1);
+        pose[2] = new EulerAngle(-ten, 0, -ten);
+        pose[3] = new EulerAngle(-fifteen, 0, ten);
+        pose[4] = new EulerAngle(-one, 0, -one);
+        pose[5] = new EulerAngle(one, 0, one);
```
Comment (Paulomart) [inline src/main/java/net/glowstone/entity/objects/GlowArmorStand.java]: Maybe make them a static final constant?

File: src/main/java/net/glowstone/entity/objects/GlowArmorStand.java [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+    private boolean isVisible = true;
+    private boolean isSmall = false;
+    private boolean hasBasePlate = true;
+    private boolean hasGravity = true;
+    private boolean hasArms = false;
+
+    public GlowArmorStand(Location location) {
+        super(location);
+        setMaxHealth(2);
+        setHealth(2);
```
Comment (hibo98) [inline src/main/java/net/glowstone/entity/objects/GlowArmorStand.java]: @mastercoms When we pull #197 first then here can be used the new super constructor for GlowLivingEntity

**Comment by mastercoms** [comment]
Waiting for #197 before reviewing this.

**Comment by mastercoms** [comment]
#197 has been pulled.

**Comment by mastercoms** [comment]
Thank you for your contribution!

**Comment by mastercoms** [comment]
I found that armor stands should only break when a player hits it twice quickly. This PR unfortunately does not do that.

**Comment by kaenganxt** [comment]
It needs two hits if the player is in survival mode, but only one in creative mode. For me, it works fine in this PR.

**Comment by hibo98** [comment]
It works for me too.

**Comment by mastercoms** [comment]
Ah, I was not aware that creative mode took one hit. Never mind then.

**Pull request merged**